php/actions/authentication/authactions.php

```diff
@@ -57,14 +56,14 @@ function RegisterUser($username, $password){
 		return "USERNAME_ALREADY_REGISTERED";
 	}
 
-	$salt = GenerateSalt();
-	$passwordIterations = GenerateUserHashIterations($configData);
-	$passwordHash = HashPassword($password, $salt, $passwordIterations, $configData);
 	$isAdmin = (count($userData->UserModels) == 0) ? 1 : 0;
 
-	$userDbInterface->Insert($username, $ip, $userAgent, $salt, $passwordHash, AUTH_VERSION, $passwordIterations, $isAdmin);
-	
+	// create user without password
+	$userDbInterface->Insert($username, $ip, $userAgent, OVERRIDE_UNUSED, OVERRIDE_UNUSED, AUTH_BCRYPT, OVERRIDE_UNUSED, $isAdmin);	
+
 	$userData = new UserData($userDbInterface, $sessionDbInterface, $configData);
+	// change the new user's password to the one provided by the user;
+	UpdateUserPassword($userData->UsernameToId[$username], $password, true);
```

Comment:
Imo it'd probably be best to generate the password ahead of time and provide it with the `Insert` function, as there's a possibility something would break in or between `Insert()` and `UpdateUserPassword()`, causing a half-created user. This isn't critical to fix, but we should probably create a ticket so it gets fixed later.